spec/system/consumer/split_checkout_spec.rb

```diff
@@ -207,10 +207,109 @@
         end
       end
 
-      describe "filling out delivery details" do
+      describe "filling out order details" do
         before do
           fill_out_details
           fill_out_billing_address
+          order.update(user_id: user.id)
+        end
+
+        context "billing address" do
+          describe "checking the default address box" do
+            before do
+              check "order_save_bill_address"
+              choose free_shipping.name
+            end
+
+            it "before saving has no default address" do
+              expect(user.bill_address).to be_nil
+              expect(order.bill_address).to be_nil
+            end
+
+            context "as a first time customer" do
+              before do
+                proceed_to_payment
+              end
+              it "creates a new default bill address" do
+                expect(order.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(order.customer.bill_address.address1).to eq "Rue de la Vie, 77"
+                expect(user.reload.bill_address.address1).to eq "Rue de la Vie, 77"
+              end
+            end
+          end
+
+          describe "unchecking the default address box" do
+            before do
+              uncheck "order_save_bill_address"
+              choose free_shipping.name
+            end
+
+            context "as an existing customer" do
+              let(:existing_address) { create(:address) }
+
+              before do
+                user.bill_address = existing_address
```

Comment:
Hum, interesting - so the idea would be to store the previous user.address, and not define another one, right? I think Rspec treats each context/example separately, so the the only strategy which came to mind was creating a new address, with the factory.

Another option, could be nesting this context "as an existing customer" in the previous one could perhaps skip the need to assign another address, but it would make the second test dependent on the first one... Not sure this is what you mean.